go-controller/pkg/network-controller-manager/network_controller_manager.go

```diff
@@ -380,7 +381,19 @@ func (cm *NetworkControllerManager) Start(ctx context.Context) error {
 	}
 	cm.podRecorder.Run(cm.sbClient, cm.stopChan)
 
-	err = cm.initDefaultNetworkController()
+	var observabilityManager *observability.Manager
+	if config.OVNKubernetesFeature.EnableObservability {
+		observabilityManager = observability.NewManager(cm.nbClient)
+		if err = observabilityManager.Init(); err != nil {
+			return fmt.Errorf("failed to init observability manager: %w", err)
+		}
+	} else {
+		err = observability.Cleanup(cm.nbClient)
+		if err != nil {
+			klog.Warningf("Observability cleanup failed, expected if not all Samples ware deleted yet: %v", err)
+		}
+	}
+	err = cm.initDefaultNetworkController(observabilityManager)
```

Comment:
Shouldn't we pass this same observabilityManager to all secondary network controllers (for user defined networks)?  IIUC user defined networks support network policy, we might as well make them "observable".

Reply:
I was going to to that in the next release, as I think secondary network observability may need some more data, e.g. to which secondary network given network policy belongs, and also I didn't test this functionality yet.
I think I should just document this limitation for now and make sure I add this feature later.